meson.build

```diff
@@ -0,0 +1,350 @@
+## SPDX-License-Identifier: 0BSD
+## Author: Vincent Torri
+
+project(
+    'xz',
+    'c',
+    version: '5.6.99',
+    license: '0BSD',
+    license_files: [
+        'COPYING',
+        'COPYING.0BSD',
+        'COPYING.GPLv2',
+        'COPYING.GPLv3',
+        'COPYING.LGPLv2.1'
+    ],
+    default_options: [
+        'b_ndebug=if-release',
+        'c_std=c11',
+        'warning_level=3',
+    ],
+    meson_version: '>= 1.1.0',
+)
+
+v_array = meson.project_version().split('.')
+v_maj = v_array[0]
+
+# config.h
+
+config_dir = [include_directories('.')]
+
+config_h = configuration_data()
+config_h.set_quoted('PACKAGE', 'xz')
+config_h.set_quoted('PACKAGE_BUGREPORT', 'xz@tukaani.org')
+config_h.set_quoted('PACKAGE_NAME', meson.project_name())
+config_h.set_quoted('PACKAGE_URL', 'https://tukaani.org/xz/')
+
+# host
+
+config_h.set10('WORDS_BIGENDIAN', host_machine.endian() == 'big',
+    description: 'Define to 1 if the processor stores words with the most significant byte first.'
+)
```

Comment:
`config_h.set10` always defines the macro to 1 or 0. The C code uses `#ifdef WORDS_BIGENDIAN`. Thus now it always thinks that it's on big endian. On little endian systems this results in liblzma that can decompress the files it creates (so certain tests will pass) but the files will be wrong. Testing with the included test files catches this (like the testing done by `test_files.sh`), thus this is a clear example why the source package needs test files.

`set10` is a good match when converting certain Autoconf macros like `AC_CHECK_DECLS` which always define the C macro to 1 or 0. Those feel a bit odd and we got rid of them in XZ Utils some time ago. I think Meson build of XZ Utils shouldn't use `set10` at all. Suggested code:

```
if host_machine.endian() == 'big'
    config_h.set('WORDS_BIGENDIAN', 1,
        description: 'Define to 1 if the processor stores words with the most significant byte first.')
endif
```

Reply:
Correction: Due to the use of the unmodified visibility.m4 from Gnulib, `HAVE_VISIBILITY` has to be always defined and thus use `set10`.

Reply:
I have always felt the opposite regarding weirdness. If you use ifdef, you can't tell whether files are supposed to not have it defined or whether the build system forgot to check it / config.h wasn't included in that file. If it is always guaranteed to be defined, and you check which state it has with `#if`, then you can take advantage of -Wundef.

Still, the port must accommodate the code that exists, not the code that contributors would prefer...

Reply:
I agree. I meant weird in context of Autoconf where almost all other checks either define or not define a macro instead of always defining to 1 or 0. I don't like mixing both styles because due to `-Wundef` one cannot use `#if` to check those macros that aren't always defined. When both styles are present, one has to be careful to know which macros to check with `#ifdef` and which with `#if`. Nowadays `HAVE_VISIBILITY` is such a speciality in XZ Utils.

Converting the codebase to only use macros that are always defined is possible but it's quite a bit of work too and there is a risk of errors. It would also affect build systems that are maintained outside the project. Maybe it could be done some day, who knows, but it's not a high priority for now.

Reply:
Autoconf's `AC_DEFINE` always puts something into config.h, either `#define FOO 1` or `/* #undef FOO */`. This way it's easy to see both what was and what wasn't defined.

Meson cannot do that. Thus in Meson context the 1/0 method has an extra benefit on top of what was already mentioned.

Reply:
With set() instead of set10, it expects either true or false and true will `#define FOO` but not set it to 1 -- false will #undef it.

Reply:
Ah, thanks! Now I wonder which I prefer. The explicit `#undef` has a clearer meaning than Autoconf's commented version, but Autoconf's version is visually easier to spot in the list. The explicit one is trivial to filter to something else _for viewing purposes_ so it's not a problem. :-)

I think I've been careful to never assume that a macro has a value excluding the special cases. And an empty value in `#if` would usually be an error. So perhaps the `set` method with true or false is nicer. It avoids `if`...`endif` clutter in `meson.build` files.

Reply:
so is

```
config_h.set('WORDS_BIGENDIAN', host_machine.endian() == 'big',
    description: 'Define to 1 if the processor stores words with the most significant byte first.'
)
```

(which generates `#undef WORDS_BIGENDIAN` on little endian) good for you ?

Reply:
done in fb6b458e9ed311ed1728de3b4c4617fd8d5861d3